### PR TITLE
dockerfile: optimize dockerfile

### PR DESCRIPTION
**Pull request by purelind**
### What problem does this PR solve?

Use the official golang mirror as the builder image, supporting building on Linux arm64.
Issue Number: close https://github.com/pingcap/tidb/issues/49071,https://github.com/pingcap/tidb/issues/49560

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,22 +13,16 @@
 # limitations under the License.
 
 # Builder image
```
Comment (wuhuizuo) [inline Dockerfile:15]: It's better add comment in the Dockerfile and README.md to notice it just for development.
And the product image Dockerfile in managed in:
- just the final image: https://github.com/PingCAP-QE/artifacts/blob/main/dockerfiles/products/tidb.Dockerfile
- multi stages dockerfile: https://github.com/PingCAP-QE/artifacts/blob/main/dockerfiles/cd/builders/tidb/Dockerfile

Comment (purelind) [reply Dockerfile:18]: Some information has been added.

**Review by Defined2014 — APPROVED** [review]
Please also link https://github.com/pingcap/tidb/issues/49071, reset LGTM

**Review by wuhuizuo — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**